instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryConnectionTest.java

```diff
@@ -0,0 +1,235 @@
+    String query = "SELECT * FROM users";
+    Statement statement = connection.createStatement();
+
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          assertThat(statement.execute(query)).isTrue();
+        });
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName("SELECT my_name.users")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(DbIncubatingAttributes.DB_SYSTEM, dbInfo.getSystem()),
+                            equalTo(DbIncubatingAttributes.DB_NAME, dbInfo.getName()),
+                            equalTo(DbIncubatingAttributes.DB_USER, dbInfo.getUser()),
+                            equalTo(DbIncubatingAttributes.DB_STATEMENT, query),
+                            equalTo(DbIncubatingAttributes.DB_OPERATION, "SELECT"),
+                            equalTo(DbIncubatingAttributes.DB_SQL_TABLE, "users"),
+                            equalTo(ServerAttributes.SERVER_ADDRESS, dbInfo.getHost()),
+                            equalTo(ServerAttributes.SERVER_PORT, dbInfo.getPort()))));
+
+    statement.close();
+    connection.close();
+  }
+
+  @Test
+  void testVerifyCreateStatementReturnsOtelWrapper() throws Exception {
+    OpenTelemetry ot = OpenTelemetry.propagating(ContextPropagators.noop());
+    Instrumenter<DbRequest, Void> instrumenter = createStatementInstrumenter(ot);
+    OpenTelemetryConnection connection =
+        new OpenTelemetryConnection(new TestConnection(), DbInfo.DEFAULT, instrumenter);
+
+    assertThat(connection.createStatement()).isInstanceOf(OpenTelemetryStatement.class);
+    assertThat(connection.createStatement(0, 0)).isInstanceOf(OpenTelemetryStatement.class);
+    assertThat(connection.createStatement(0, 0, 0)).isInstanceOf(OpenTelemetryStatement.class);
+
+    Field instrumenterField = OpenTelemetryStatement.class.getDeclaredField("instrumenter");
+    instrumenterField.setAccessible(true);
+
+    assertThat(instrumenterField.get(connection.createStatement())).isEqualTo(instrumenter);
+
+    connection.close();
+  }
+
+  @Test
+  void testVerifyPrepareStatement() throws SQLException {
+    Instrumenter<DbRequest, Void> instrumenter =
+        createStatementInstrumenter(testing.getOpenTelemetry());
+    DbInfo dbInfo = getDbInfo();
+    OpenTelemetryConnection connection =
+        new OpenTelemetryConnection(new TestConnection(), dbInfo, instrumenter);
+    String query = "SELECT * FROM users";
+    PreparedStatement statement = connection.prepareStatement(query);
+
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          assertThat(statement.execute()).isTrue();
+        });
+
+    testing.waitAndAssertTraces(
```

Comment:
Here you can extract a public method, and then simplify the code together with the following verification logic. Can refer to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/1823f147eb0c18ac6259827f16d7808a7dc6774e/instrumentation/xxl-job/xxl-job-common/testing/src/main/java/io/opentelemetry/instrumentation/xxljob/AbstractXxlJobTest.java#L165